docs/messaging-implementation.md

```diff
@@ -60,12 +67,15 @@ Widget messaging is based on JSON input configured in a [widget's configuration]
 - **priority**: Set to "high" if you want the message to be displayed with higher visibility (i.e. As a priority notification or popup announcement, respectively).
 - **recurrence**:*(experimental, optional)* If true, even if a notification is dismissed, it will continue to reoccur in the user's home at the start of every session until the user is no longer a member of the targeted group. For example, if a user is a member of students-with-outstanding-parking-tickets, that user will be confronted with the notification at every login until they pay the fine.
 - **dismissible**: *(experimental, optional)* Set to false if you want to disallow users from dismissing a notification. This should only be used for truly critical messages. If the attribute is set to true or not set at all, the notification will be dismissible.
-- **audienceFilter**: A group of attributes related to filtering messages based on a user's group or whether the user has pertinent data at a given URL.
+- **audienceFilter**: A group of attributes related to filtering messages based on a user's group.
   - **groups**: An attribute to optionally show messages only to specific groups (i.e. uPortal groups). If null or empty array, the message will be shown to everyone. Contact your portal development team for more information about group filtering.
+- **data**: *(optional)* A group of attributes related to external data retrieved by a dataUrl.
   - **dataUrl**: *(optional)* The message will retrieve data from the dataUrl. If data exists, it will show this message to the user. Only supports JSON.
     You would use this feature if you want to only show the message if the specific user has data. For example: Only show user if they have a certain document.
   - **dataObject**: *(optional)* Will only be looked at if `dataUrl` is present, otherwise ignored. Used as an optional further refinement from dataUrl, if you want the notification to show only if the specific object is in the data.
-  - **dataArrayFIlter**: Will only be looked at if `dataUrl` is present, otherwise ignored. Used as an optional further refinement from dataUrl. If your object return is an array, you can filter on the array. Does support multiple filtering criteria as shown in the example. If used in conjunction with `dataObject`, will filter to `dataObject` first.  [AngularJS array filtering documentation] (https://docs.angularjs.org/api/ng/filter/filter)
+  - **dataArrayFilter**: Will only be looked at if `dataUrl` is present, otherwise ignored. Used as an optional further refinement from dataUrl. If your object return is an array, you can filter on the array. Does support multiple filtering criteria as shown in the example. If used in conjunction with `dataObject`, will filter to `dataObject` first.  [AngularJS array filtering documentation] (https://docs.angularjs.org/api/ng/filter/filter)
```

Comment:
:thinking: :question: with the `data` object wrapping, is the `data` prefix on the property still wanted?

Reply:
Yes.  I think the names aren't that great to begin (I created them, so I don't feel too bad saying that) 
 and without the data prefix, they might even be worse.  We need some new names here.

Reply:
Relevant, great quote

> There are only two hard things in Computer Science: cache invalidation and naming things.
> -- Phil Karlton

source: https://www.martinfowler.com/bliki/TwoHardThings.html